### PR TITLE
Rewrote creating games for tests yet again.

### PR DESCRIPTION
**Pull request by kberg**
I have mixed feelings about this approach. It introduces a fake form of extension that doesn't really accomplish what I set out to do, which is to provide an equivalent to TestPlayer. The difference is that TestPlayers' parent, Player, has an accessible constructor, but Game does not.

So, just not clear.

### CODE REVIEW

**Comment by kberg** [comment]
@bafolts I'd really like your thoughts about this approach.

File: tests/TestGame.ts [hunk 1]
```diff
@@ -18,13 +20,9 @@ export function newTestGame(count: number, customOptions?: Partial<GameOptions>,
   const options: GameOptions | undefined = customOptions === undefined ?
     undefined :
     setCustomGameOptions(customOptions);
-  return Game.newInstance(`game-id${idSuffix}`, players, players[0], options);
-}
-
-export function getTestPlayer(game: Game, idx: number): TestPlayer {
-  return game.getPlayers()[idx] as TestPlayer;
-}
+  const game = Game.newInstance(`game-id${idSuffix}`, players, players[0], options);
```
Comment (bafolts) [inline tests/TestGame.ts:23]:
```suggestion
  const game: TestGame = Game.newInstance(`game-id${idSuffix}`, players, players[0], options);
  game.testPlayers = players;
  return game;
```

Save an _any_ and be proud!

**Review by bafolts — APPROVED** [review]
This approach seems fine. With my suggestion the _any_ can be avoided.

File: tests/TestGame.ts (hunk 1 above)
Comment (kberg) [reply tests/TestGame.ts:23]: Unfortunately this doesn't work.
image

Comment (bafolts) [reply tests/TestGame.ts:23]: When I make the change locally the build works for me. Is this error seen while building? I suggested the change after being able to build locally.

**Comment by kberg** [comment]
I'm gonna close this one for now.

**Pull request closed without merging**